cmd/subctl/recover_broker_info.go

```diff
@@ -0,0 +1,175 @@
+			if err != nil {
+				return err
+			}
+
+			clusters, err := brokerCluster.GetClusters(brokerNamespace)
+			if err != nil {
+				return status.Error(err, "error listing joined clusters")
+			}
+
+			ok, err := tryToRecoverFromBroker(brokerCluster, brokerObj, brokerNamespace, clusters, status)
+			if ok || err != nil {
+				return err
+			}
+
+			status.Warning("Submariner is not installed on the same cluster as Broker")
+			status.Start("Checking if Submariner is installed on a different cluster")
+			//nolint:wrapcheck // No need to wrap errors here.
+			return recoverRestConfigProducer.RunOnSelectedContext(
+				func(submCluster *cluster.Info, namespace string, status reporter.Interface) error {
+					if isSubmJoinedToBroker(clusters, submCluster) {
+						status.Success("Found a Submariner installation on cluster %q joined to the Broker", submCluster.Name)
+						//nolint:wrapcheck // No need to wrap errors here.
+						return broker.RecoverData(brokerCluster, submCluster, brokerObj, namespace, status)
+					}
+
+					return status.Error(
+						fmt.Errorf("submariner is not installed on cluster %s. "+
+							"Please specify the cluster where Submariner is installed via `--kubeconfig` or `--context` flag"+
+							"", submCluster.Name), "")
+				}, status)
+		}, status)
+
+	if contextFound || err != nil {
+		return err //nolint:wrapcheck // No need to wrap errors here.
+	}
+
+	// if --brokerconfig not provided, search for broker on current context and proceed to get Submariner
+	//nolint:wrapcheck // No need to wrap errors here.
+	return recoverRestConfigProducer.RunOnSelectedContext(
+		func(clusterInfo *cluster.Info, namespace string, status reporter.Interface) error {
+			brokerObj, err := getBroker(clusterInfo, namespace, "Please specify the cluster where the Broker is installed "+
+				"via the "+"`--brokerconfig` or `--brokercontext` flag and the namespace via the --brokernamespace flag", status)
+			if err != nil {
+				return err
+			}
+
+			clusters, err := clusterInfo.GetClusters(namespace)
+			if err != nil {
+				return status.Error(err, "error listing joined clusters")
+			}
+
+			ok, err := tryToRecoverFromBroker(clusterInfo, brokerObj, namespace, clusters, status)
+			if ok || err != nil {
+				return err
+			}
+
+			return status.Error(
+				fmt.Errorf("submariner is not installed on cluster %q. "+
+					"Please specify the cluster where Submariner is installed via `--kubeconfig` or `--context` flag"+
+					"", clusterInfo.Name), "")
+		}, status)
+}
+
+func getBroker(clusterInfo *cluster.Info, namespace, notFoundMsg string, status reporter.Interface) (*v1alpha1.Broker, error) {
+	status.Start("Checking if the Broker is installed on cluster %q in namespace %q", clusterInfo.Name, namespace)
+
+	brokerObj, err := clusterInfo.GetBroker(namespace)
+	if apierrors.IsNotFound(err) {
+		return nil, status.Error(fmt.Errorf("the Broker is not installed on the specified cluster. %s", notFoundMsg), "")
```

Comment:
You should also specify namespace as it could be significant for the failure info